apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/DcGeoRDALoadJob.java

```diff
@@ -0,0 +1,149 @@
+package gov.cms.bfd.pipeline.rda.grpc;
+
+import static gov.cms.bfd.pipeline.sharedutils.PipelineJobOutcome.NOTHING_TO_DO;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import gov.cms.bfd.pipeline.sharedutils.PipelineJob;
+import gov.cms.bfd.pipeline.sharedutils.PipelineJobOutcome;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Skeleton PipelineJob instance that delegates the actual ETL work to two other objects. The
+ * RDASource object handles communication with the source of incoming data. The RDASink object
+ * handles communication with the ultimate storage system. The purpose of this class is to handle
+ * general PipelineJob semantics that are common to any source or sink.
+ */
+public final class DcGeoRDALoadJob implements PipelineJob {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DcGeoRDALoadJob.class);
+  public static final String SCAN_INTERVAL_PROPERTY = "DCGeoRDALoadIntervalSeconds";
+  public static final String RUN_TIME_PROPERTY = "DCGeoRDALoadRunSeconds";
+  public static final String MAX_RECORDS_PROPERTY = "DCGeoRDALoadMaxRecords";
+  public static final int SCAN_INTERVAL_DEFAULT = 300;
+  public static final int RUN_TIME_DEFAULT = 300;
+  public static final int MAX_RECORDS_DEFAULT = Integer.MAX_VALUE;
+  public static final String BATCH_SIZE_PROPERTY = "DCGeoBatchSize";
+  public static final int BATCH_SIZE_DEFAULT = 1;
+  public static final String CALLS_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "calls");
+  public static final String FAILURES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "failures");
+  public static final String SUCCESSES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "successes");
+  public static final String PROCESSED_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "processed");
+
+  private final Config config;
+  private final Callable<RDASource<PreAdjudicatedClaim>> sourceFactory;
+  private final Callable<RDASink<PreAdjudicatedClaim>> sinkFactory;
+  private final Meter callsMeter;
+  private final Meter failuresMeter;
+  private final Meter successesMeter;
+  private final Meter processedMeter;
+
+  public DcGeoRDALoadJob(
+      Config config,
+      Callable<RDASource<PreAdjudicatedClaim>> sourceFactory,
+      Callable<RDASink<PreAdjudicatedClaim>> sinkFactory,
```

Comment:
It's best to consider the PreAdjudicatedClaim object as a root level object containing whatever objects our schema requires.  It might actually contain a set of objects of different types or something (we'll know that when we get to the sprint that defines our schema).  RDASource is a higher level abstraction than just gRPC/RDA.  In the gRPC client case there will be an internal mapping between whatever RDA returns to us and whatever our database schema requires.

I originally left PreAdjudicatedClaim as a generic type too but thought it best to make a concrete class here to avoid confusion and prevent the PipelineManager having to deal with a generic type.